common/src/api/internal/shared.rs

```diff
@@ -1101,6 +1101,50 @@ pub struct SledIdentifiers {
     pub serial: String,
 }
 
+/// Identifies the kind of CPU present on a sled, determined by reading CPUID.
+///
+/// This is intended to broadly support the control plane answering the question
+/// "can I run this instance on that sled?" given an instance with either no or
+/// some CPU platform requirement. It is not enough information for more precise
+/// placement questions - for example, is a CPU a high-frequency part or
+/// many-core part? We don't include Genoa here, but in that CPU family there
+/// are high frequency parts, many-core parts, and large-cache parts. To support
+/// those questions (or satisfactorily answer #8730) we would need to collect
+/// additional information and send it along.
+#[derive(
+    Serialize, Deserialize, Copy, Clone, Debug, PartialEq, Eq, JsonSchema,
+)]
+#[serde(rename_all = "snake_case")]
+pub enum SledCpuFamily {
+    /// The CPU vendor or its family number don't correspond to any of the
+    /// known family variants.
+    Unknown,
+
+    /// AMD Milan processors (or very close). Could be an actual Milan in a
+    /// Gimlet, a close-to-Milan client Zen 3 part, or Zen 4 (for which Milan is
+    /// the greatest common denominator).
+    AmdMilan,
+
+    /// AMD Turin processors (or very close). Could be an actual Turin in a
+    /// Cosmo, or a close-to-Turin client Zen 5 part.
+    AmdTurin,
+
+    /// AMD Turin Dense processors. There are no "Turin Dense-like" CPUs unlike
+    /// other cases, so this means a bona fide Zen 5c Turin Dense part.
+    AmdTurinDense,
+}
+
+impl fmt::Display for SledCpuFamily {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            SledCpuFamily::Unknown => write!(f, "unknown"),
+            SledCpuFamily::AmdMilan => write!(f, "milan"),
+            SledCpuFamily::AmdTurin => write!(f, "turin"),
+            SledCpuFamily::AmdTurinDense => write!(f, "turin_dense"),
```

Comment:
Note (i'm not sure if you care) that this will format slightly differently than what Serde will serialize the enum as, since serde's formatting will be prefixed with `amd_` for `AmdMilan`, `AmdTurin`, and `AmdTurinDense` --- do we care about making these the same?

Also, a super unimportant nit: for enums like this, I often like to add a `pub fn as_str(&self) -> &'static str` and then just make `Display` use that, so there's a way to get the static string out without allocating/formatting, in the handful of cases where you might just need an `&str`. Not necessary here if you're not using that, but it's just a thing I like to do so it's there when you want it and people don't `.to_string()` the enum.

Reply:
these should have the `amd_` prefix, yeah. the codenames are seared into my brain as "AMD" at this point so i didn't even realize i'd omitted it